src/nrf_802154.h

```diff
@@ -1002,7 +1002,31 @@ void nrf_802154_pan_coord_set(bool enabled);
 bool nrf_802154_pan_coord_get(void);
 
 /**
- * @brief Adds the address of a peer node for which the provided ACK data is to be set.
+ * @brief Select the source matching algorithm.
+ *
+ * @note This method should be called after driver initialization, but before transceiver is enabled.
+ *
+ * When calling @ref nrf_802154_ack_data_pending_bit_should_be_set, one of several algorithms
+ * for source address matching will be chosen. To ensure a specific algorithm is selected,
+ * call this function before @ref rf_802154_ack_data_pending_bit_should_be_set.
+ *
+ * @param[in]  match_method Source address matching method to be used.
+ */
+void nrf_802154_src_addr_matching_method_set(nrf_802154_src_addr_match_t match_method);
+
+/**
+ * @brief Adds the address of a peer node for which the provided ACK data
+ * is to be added to the pending bit list.
+ *
+ * The pending bit list works differently, depending on the upper layer for which the source
+ * address matching method is selected.

```

Comment:
```suggestion
* address matching method is selected:
```
Mirror the edit elsewhere in the PR.